### ISSUE TITLE
False Positive when using function descriptors 

### ISSUE THREAD

**Issue by BlueMagma2** [PyCQA/pylint#7736]
### Bug description

If I want to create a classmethod from a function on an existing class, pylint wrongfully report an error

Take the following code example
```
def my_func(cls, my_list):
    if len(my_list) == 0:
        my_list.append(1)

class MyClass:
    def __init__(self):
        pass

MyClass.my_class_method = my_func.__get__(MyClass)

a_list = []
MyClass.my_class_method(a_list)
print(a_list)
```

### Command used

```shell
python3.8 -m pylint -rn -f colorized --errors-only --reports=y example.py
```

### Pylint output

```shell
example.py:9:0: E1111: Assigning result of a function call, where the function has no return (assignment-from-no-return)
example.py:9:26: E1120: No value for argument 'my_list' in function call (no-value-for-parameter)
example.py:9:26: E1120: No value for argument 'type' in function call (no-value-for-parameter)
```

### Expected behavior

No error should be returned

### Pylint version

```shell
2.15.5
```

### OS / Environment

Ubuntu 20.04.5 LTS

**Comment by nickdrozd** [comment]
This kind of extravagantly dynamic behavior is exactly what makes Python so fun to write and so frustrating to statically analyze. It's probably too clever for Pylint, and at any rate it's too clever for me. Unless anyone has any better ideas, I suggest adding a `pylint: disable` comment and charging it to the game.
